packages/components/src/internal/components/forms/DomainFieldMetadata.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { FC } from 'react';
+
+import { QueryColumn } from '../../../public/QueryColumn';
+
+export interface Props {
+    column?: QueryColumn;
+    description?: string;
+    required?: boolean;
+    type?: string;
+}
+
+export const DOMAIN_FIELD = 'DomainField';
+
+export const DomainFieldMetadata: FC<Props> = props => {
```

Comment:
minor: the word "metadata" in this component name throw me off for a bit. Thoughts on something like DomainFieldHelpTipContents or something along those lines? I don't have a real good suggestion.

Reply:
Yeah, I struggled with the name a bit, so open to suggestions. Including HelpTip seemed somewhat redundant, but 🤷